src/gui/guiConfirmRegistration.h

```diff
@@ -0,0 +1,59 @@
+/*
+Part of Minetest
+Copyright (C) 2017 srifqi, Muhammad Rifqi Priyo Susanto
+		<muhammadrifqipriyosusanto@gmail.com>
+
+Permission to use, copy, modify, and distribute this software for any
+purpose with or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#pragma once
+
+#include "irrlichttypes_extrabloated.h"
+#include "modalMenu.h"
+#include <string>
+
+class Client;
+
+class GUIConfirmRegistration : public GUIModalMenu
+{
+public:
+	GUIConfirmRegistration(gui::IGUIEnvironment *env, gui::IGUIElement *parent,
+			s32 id, IMenuManager *menumgr, Client *client,
+			const std::string &playername, const std::string &password,
+			const std::string &address, bool *aborted);
+	~GUIConfirmRegistration();
+
+	void removeChildren();
+	/*
+		Remove and re-add (or reposition) stuff
+	*/
+	void regenerateGui(v2u32 screensize);
+
+	void drawMenu();
+
+	void closeMenu(bool goNext);
+
+	void acceptInput();
+
+	bool processInput();
+
+	bool OnEvent(const SEvent &event);
+
+private:
+	Client *m_client;
+	const std::string &m_playername;
+	const std::string &m_password;
+	const std::string &m_address;
+	bool *m_aborted;
```

Comment:
Could you please initialize the pointers here for safety, if the code is altered later?

Reply:
Which pointer?

Reply:
`bool *m_aborted` and `Client *m_client`. Please set them to `nullptr` here.